srtcore/handshake.h

```diff
@@ -263,10 +263,11 @@ inline int RejectReasonForURQ(UDTRequestType req)
     if (req < URQ_FAILURE_TYPES)
         return SRT_REJ_UNKNOWN;
 
-    if (req < URQ_SERVER_FAILURE_TYPES && req - URQ_FAILURE_TYPES >= SRT_REJ_E_SIZE)
+    int reason = req - URQ_FAILURE_TYPES;
+    if (reason < SRT_REJC_SERVER && reason >= SRT_REJ_E_SIZE)
```

Comment:
I may assume `reason >= SRT_REJ_E_SIZE` already covers `reason < SRT_REJC_SERVER`

Reply:
Excuse me?

It simply maps to `reason >= 16` and it should cover `reason < 1000` ? How about the values of `reason >= 1000` ?

Reply:
In short: the rule here is that:
 - for rejection codes between 1 and `SRT_REJ_E_SIZE-1` the rejection value is returned
 - for values from `SRT_REJC_SERVER` on, the rejection value is returned (even if this results in something stupid, like an nonexistent HTTP code)
 - otherwise (0 and between `SRT_REJ_E_SIZE` and `SRT_REJCT_SERVER-1`) it should return UNKNOWN

Reply:
Right, I see what you are doing here now.
You are returning `SRT_REJ_UNKNOWN` for those values, that are right after `SRT_REJ_E_SIZE` (including) and up to first user-defined code `SRT_REJC_SERVER`.

Reply:
Exactly. This is returned for UNKNOWN itself and for undefined codes in the SRT internal range only. In the server and user ranges it is returned as is.